### ISSUE TITLE
'nodeIP' option for openshift node configuration

### ISSUE THREAD

**Issue by pravisankar** [openshift/openshift-ansible#549]
If 'nodeIP' option is provided, this IP will be used for pod traffic communication.
PR: https://github.com/openshift/origin/pull/4459 (pending merge)
Once the pull request is merged, please update your templates.

**Comment by liggitt** [comment]
If this is set, I think the nodeIP should also be included in the node serving certificate (by adding it to the `--hostnames` arg in `create-node-config`)

**Comment by detiber** [comment]
@liggitt Are we setting the SAN attributes based on the type of value being passed in (IP vs DNS) or are we just setting them all as Alt DNS entries?

**Comment by liggitt** [comment]
We do insane things to make certs that work everywhere.

1. We divide passed `--hostnames` into DNS and IP entries
2. We add all IP addresses as IP SANs
3. We add all DNS names and IP addresses as DNS SANs, with DNS names coming first

IP addresses have to be added as DNS SANs to make Windows happy, but have to come last to avoid making Firefox unhappy (see https://bugzilla.redhat.com/show_bug.cgi?id=1218486#c2)

**Comment by detiber** [comment]
@sdodson while we are updating certs for users during the upgrade, we'll probably also want to address this as well...  we should be passing in all configured dns names (all_hostnames) unioned with  a new aggregated variable for ips as well (all_ips maybe, where we gather all possibly IP address values for a service/host) into the --hostnames field when creating the node certs.

**Comment by tbielawa** [comment]
This issue has been inactive for quite some time. Please update and reopen this issue if this is still a priority you would like to see action on.

**Comment by tbielawa** [comment]
Reopened for @liggitt to get verification

**Comment by michaelgugino** [comment]
The linked bugzilla document shows that issue closed and fixed in recent firefox versions.

If this is still affecting you, please file a new issue.
